src/query/sensor/processors/sensor.processor.ts

```diff
@@ -21,38 +21,45 @@ export class SensorProcessor {
   ) {
   }
 
+  private errorCallback(error: any): void {
+    if (error) {
+      this.logError(event);
+    }
+  }
+
   protected logger: Logger = new Logger(this.constructor.name);
 
   async process(event): Promise<void> {
 
+    let result: ISensor;
     if (event instanceof SensorRegistered) {
-      await this.processCreated(event);
+      result = await this.processCreated(event);
     } else if (event instanceof SensorUpdated) {
-      await this.processUpdated(event);
+      result = await this.processUpdated(event);
     } else if (event instanceof SensorDeleted) {
-      await this.processDeleted(event);
+      result = await this.processDeleted(event);
     } else if (event instanceof SensorActivated) {
-      await this.processActivated(event);
+      result = await this.processActivated(event);
     } else if (event instanceof SensorDeactivated) {
-      await this.processDeactivated(event);
+      result = await this.processDeactivated(event);
     } else if (event instanceof SensorOwnershipShared) {
-      await this.processOwnershipShared(event);
+      result = await this.processOwnershipShared(event);
     } else if (event instanceof SensorOwnershipTransferred) {
-      await this.processOwnershipTransferred(event);
+      result = await this.processOwnershipTransferred(event);
     } else if (event instanceof DatastreamAdded) {
-      await this.processDataStreamCreated(event);
+      result = await this.processDataStreamCreated(event);
     } else if (event instanceof DatastreamDeleted) {
-      await this.processDataStreamDeleted(event);
+      result = await this.processDataStreamDeleted(event);
     } else if (event instanceof SensorRelocated) {
-      await this.processLocationUpdated(event);
+      result = await this.processLocationUpdated(event);
     } else {
       this.logger.warn(`Caught unsupported event: ${event}`);
     }
 
-    this.sensorGateway.emit(event.constructor.name, event);
+    this.sensorGateway.emit(event.constructor.name, result.toObject());
```

Comment:
Hmm... interessante benadering. Ik weet niet wat ik er van vind. Is het juist als een `processEvent` method ook de verantwoordelijkheid heeft om sensor data te retourneren? Of zou dat in een aparte class moeten gebeuren? Tegelijk héb je daar precies die data ... dus is het wel logisch ... 🤔 

Het is in ieder geval veel beter dan de events emitten!! 👍

Reply:
Dit is al beter dan wat het was, maar ik twijfel zelf ook of dit wel de beste plek is. Valt eigenlijk weer samen met een discussie die we eerder hadden over de saga's, intuïtief denk ik nog steeds dat dat de meest 'kloppende' plaats zal zijn voor terugkoppeling naar eindgebruikers 🤔 Voor nu zo maar laten staan? Wanneer ik een momentje over heb, wil ik wel kijken of ik die sagas kan opzetten.

Reply:
Prima!